src/components/performance-forecast/statistics.cjsx

```diff
@@ -17,20 +17,27 @@ Statistics = React.createClass
 
   propTypes:
     courseId: React.PropTypes.string.isRequired
+    roleId:   React.PropTypes.string.isRequired
     section:  ChapterSectionType.isRequired
     displaying: React.PropTypes.string.isRequired
 
-  render: ->
-    role = CurrentUserStore.getCourseRole(@props.courseId, true)
+  getWorkedText: (role) ->
     count = @props.section.clue.unique_learner_count
     total = @props.section.questions_answered_count
+    switch role
+      when 'teacher'
+        "#{pluralize(' students', count, true)}
+         #{pluralize(' has', count)} worked #{pluralize(' problems', total, true)}"
+      when 'student'
+        "#{pluralize(' problems', total, true)} worked in this #{@props.displaying}"
+      when 'teacher-student'
+        "#{pluralize(' problems', total, true)} worked"
 
-    teacherWorkedText =
-      "#{pluralize(' students', count, true)}
-       #{pluralize(' has', count)} worked #{pluralize(' problems', total, true)}"
-
-    studentWorkedText =
-      "#{pluralize(' problems', total, true)} worked in this #{@props.displaying}"
+  render: ->
+    if @props.roleId?
```

Comment:
The `roleId` is marked as required above. Either it should not be required (above) or change the places in the code that use `<Statistics>` to pass in a `teacher-student` role (slight preference for this option since it's more documented and less *"magic"*).